From dcd03838bebdece3ca1bd1505da074bb7806f50e Mon Sep 17 00:00:00 2001
From: Ding Yunliang <yunliang.ding@intel.com>
Date: Thu, 8 Jan 2015 18:33:02 +0800
Subject: [PATCH 2/2] REVERTME: disable all the platform clks post S3

Camera platform clocks are on after resume, probably
enabled by BIOS. A better long-term solution is needed.

Change-Id: Ia3bad655202b392fa47b93742cb5f5fc23f0da07
Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-16336
Signed-off-by: Ding Yunliang <yunliang.ding@intel.com>
Reviewed-on: https://android.intel.com:443/317378
---
 drivers/misc/vlv2_plat_clock.c |   22 ++++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/drivers/misc/vlv2_plat_clock.c b/drivers/misc/vlv2_plat_clock.c
index e75061a..62735c9 100644
--- a/drivers/misc/vlv2_plat_clock.c
+++ b/drivers/misc/vlv2_plat_clock.c
@@ -217,12 +217,34 @@ static const struct platform_device_id vlv2_plat_clk_id[] = {
 };
 MODULE_DEVICE_TABLE(platform, vlv2_plat_clk_id);
 
+static int vlv2_resume(struct platform_device *device)
+{
+	int i;
+
+	/* Initialize all clocks as disabled */
+	for (i = 0; i < MAX_CLK_COUNT; i++)
+		vlv2_plat_configure_clock(i, CLK_CONFG_FORCE_OFF);
+
+	return 0;
+}
+
+static int vlv2_suspend(struct platform_device *device)
+{
+	return 0;
+}
+
+static const struct dev_pm_ops vlv2_pm_ops = {
+	.suspend = vlv2_suspend,
+	.resume = vlv2_resume,
+};
+
 static struct platform_driver vlv2_plat_clk_driver = {
 	.probe = vlv2_plat_clk_probe,
 	.remove = vlv2_plat_clk_remove,
 	.id_table = vlv2_plat_clk_id,
 	.driver = {
 		.name = "vlv2_plat_clk",
+		.pm = &vlv2_pm_ops,
 		.owner = THIS_MODULE,
 	},
 };
-- 
1.7.9.5

